`{{link-to}}` doesn't accept named args for `route` and `model`

```typescript
import Component from '@glint/environment-ember-loose/glimmer-component';

export default class Foo extends Component {
  foo = 8;
}
```

```handlebars
{{#link-to route="foo" model=this.foo}}
{{/link-to}}
```

```
app/components/foo.hbs:1:1 - error TS2555: Expected at least 2 arguments, but got 1.

1 {{#link-to route="foo" model=this.foo}}
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 {{/link-to}}
  ~~~~~~~~~~~~

  node_modules/@glint/environment-ember-loose/-private/intrinsics/link-to.d.ts:9:23
    9   (args: EmptyObject, route: string, ...params: unknown[]): AcceptsBlocks<{
                            ~~~~~~~~~~~~~
    An argument for 'route' was not provided.
```

Using:

```handlebars
<LinkTo @route="foo" @model={{this.foo}}>
</LinkTo>
```

Instead fixes the problem.

--- a/packages/environment-ember-loose/-private/intrinsics/link-to.d.ts
+++ b/packages/environment-ember-loose/-private/intrinsics/link-to.d.ts
@@ -1,15 +1,4 @@
-import {
-  AcceptsBlocks,
-  DirectInvokable,
-  EmptyObject,
-  Invokable,
-} from '@glint/template/-private/integration';
-
-export type LinkToKeyword = DirectInvokable<{
-  (args: EmptyObject, route: string, ...params: unknown[]): AcceptsBlocks<{
-    default?: [];
-  }>;
-}>;
+import { AcceptsBlocks, DirectInvokable, Invokable } from '@glint/template/-private/integration';
 
 type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
   {
@@ -30,6 +19,13 @@ type LinkToArgs = RequireAtLeastOne<
   },
   'route' | 'model' | 'models' | 'query'
 >;
+
+export type LinkToKeyword = DirectInvokable<{
+  (args: LinkToArgs): AcceptsBlocks<{ default: [] }>;
+  (args: Partial<LinkToArgs>, route: string, ...params: unknown[]): AcceptsBlocks<{
+    default?: [];
+  }>;
+}>;
 
 export type LinkToComponent = new () => Invokable<
   (args: LinkToArgs) => AcceptsBlocks<{ default: [] }, HTMLAnchorElement>

--- a/packages/environment-ember-loose/__tests__/intrinsics/link-to.test.ts
+++ b/packages/environment-ember-loose/__tests__/intrinsics/link-to.test.ts
@@ -14,6 +14,50 @@ emitComponent(linkTo({}, 'index', 123));
 // @ts-expect-error: bad type for route name
 linkTo({}, 123);
 
+// {{#link-to}}
+
+{
+  const component = emitComponent(linkTo({}, 'index'));
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+{
+  const component = emitComponent(linkTo({ route: 'index' }));
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+{
+  const component = emitComponent(linkTo({ route: 'index', query: { a: 123 } }));
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+{
+  const component = emitComponent(linkTo({ route: 'index', models: [123, 'abc'] }));
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+{
+  const component = emitComponent(linkTo({ route: 'index', 'current-when': 'index' }));
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+{
+  const component = emitComponent(linkTo({ route: 'index', 'current-when': true }));
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+{
+  const component = emitComponent(
+    linkTo(
+      // @ts-expect-error: either route/model/etc or positional params are required
+      {}
+    )
+  );
+  expectTypeOf(component.blockParams.default).toEqualTypeOf<[]>();
+}
+
+// <LinkTo>
+
 {
   const component = emitComponent(LinkTo({ route: 'index', model: 123 }));
   expectTypeOf(component.element).toEqualTypeOf<HTMLAnchorElement>();

